### PR TITLE
🧪 Delegate Proxy & Runtime hacking

### PR DESCRIPTION
**Pull request by jdisho**
For more information have a look at #10

### CODE REVIEW

**Comment by freak4pc** [comment]
Awesome! Thanks or taking care of this, again :) 
About the question from #10: 
> Looks great @jdisho ! Should we wait on tests or merge as is? This is a really strong addition and I dont want to delay it for more weeks etc, but if the tests can be done in a few days it might be worth waiting to get at least some basic coverage

**Comment by jdisho** [comment]
Would love to add some tests, but I am having a hard time figuring out how to unit tests Development pods. Any help is very much appreciated.

**Comment by freak4pc** [comment]
Development pods specific are the issue? I think you can just do a `@testable import CombineCocoa`.

**Comment by jdisho** [comment]
Hey @freak4pc, I added some test for `UITableView` and `UICollectionView`. WDYT?

**Comment by freak4pc** [comment]
I'm gonna pull this into master so we can keep moving from there, seems solid enough and CI isn't actually running unless this would be merged, we can have another PR to add some more tests / clean this up, perhaps.

**Comment by freak4pc** [comment]
Looks good ! https://github.com/CombineCommunity/CombineCocoa/runs/851334969

File: Sources/DelegateProxy/ObjcDelegateProxy/ObjcDelegateProxy+Selector.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @objc(selectors:encodedReturnType:)
+    public static func selectorsOfClass(_ c: AnyClass, encodedReturnType: String) -> NSSet {
+        NSSet(set: selectors(ofClass: c, encodedReturnType: encodedReturnType))
+    }
+
+    private static func selectors(ofClass c: AnyClass, encodedReturnType: String) -> Set<Selector> {
+        var protocolsCount: UInt32 = 0
+        guard let protocolPointer = class_copyProtocolList(c, &protocolsCount) else { return .init() }
```
Comment (fpillet) [inline Sources/DelegateProxy/ObjcDelegateProxy/ObjcDelegateProxy+Selector.swift:24]: You have a memory leak here, forgot to `defer free(protocolPointer)`

Comment (freak4pc) [reply Sources/DelegateProxy/ObjcDelegateProxy/ObjcDelegateProxy+Selector.swift:24]: Yup that seems correct. @jdisho can you look into this?

**Comment by freak4pc** [comment]
This is a very old thread but I wanted to share a relatively massive problem with this that I found while trying to use it in a newer project. When using the ScrollView proxy on a UICollectionView, the flow layout delegate is never called and the layout is entirely messed up. Not too sure why it's happening, tried to debug it unsuccessfully so far.

I imagine you probably don't remember this implementation, but any chance you have an idea? @jdisho

**Comment by freak4pc** [comment]
Generally speaking I might be missing something but I'm not seeing any form of forward delegation, I think this only intercepts the original delegate but it doesn't actually forward to the original delegate if such exists, basically overriding the original delegate ?

**Comment by jdisho** [comment]
Interesting ... 🤔
I don't remember this being an issue back then, but maybe I'm wrong. Could you please share a code snippet?

**Pull request merged**